controllers/registeredservice_controller.go

```diff
@@ -109,13 +109,27 @@ func (r *RegisteredServiceReconciler) addServiceToCatalogs(ctx context.Context,
 
 	var errs []error
 	for _, sc := range catalogs.Items {
-		if rs.Spec.Constraints == nil || envtag.Match(sc.Name, rs.Spec.Constraints.Environments) {
+		constraints_match := true
```

Comment:
Don't think L112-116 is required. We can simply use the earlier condition `if rs.Spec.Constraints == nil || envtag.Match(sc.Name, rs.Spec.Constraints.Environments)` , if true add service to catalog else remove service from catalog.

Reply:
Agree it will work as well, but I will prefer to be more explicit with the boolean to show what matches means. How strong do you feel about this one?